iree/compiler/Dialect/HAL/Transforms/Passes.cpp

```diff
@@ -110,6 +111,8 @@ void buildHALTransformPassPipeline(OpPassManager &passManager,
   // Inline hal.device.switch ops and memoize their queries such that we can
   // better CSE/fold dispatch logic.
   passManager.addNestedPass<FuncOp>(createInlineDeviceSwitchesPass());
+  passManager.addNestedPass<FuncOp>(createRepeatDispatchesPass());
+  passManager.addNestedPass<FuncOp>(createLowerToCFGPass());
```

Comment:
don't use scf for this - doing a global cfg interchange here is a pretty big bit of side-effecting behavior, and I'm intending to support scf in certain places here in the future and don't want this removing them everywhere

Reply:
I see your point. The tradeoff here is compilation time and binary size. It takes much more time to compile and the binary size gets larger.

I can create a branch op for it, but the code would be more complicate as I expected. Any preference?

Reply:
doing the scf->cfg transformation across the entire program is more expensive than repeating a few standalone ops in a local transformation, so unless you are repeating 10000 times it should be fine (and if you are, then doing it as a runtime parameter is a better idea :)